src/Entities/DetailedProductEntity.php

```diff
@@ -2,25 +2,48 @@
 
 namespace Furniture\Entities;
 
+use \Furniture\Services\MeasurementConverterService;
+
 class DetailedProductEntity extends ProductEntity
 {
     private int $categoryId;
     private int $width;
     private int $height;
     private int $depth;
     private int $related;
+    private string $measurementUnit;
 
     public function jsonSerialize(): array
     {
         return [
             'categoryId' => $this->categoryId,
-            'width' => $this->width,
-            'height' => $this->height,
-            'depth' => $this->depth,
+            'width' => MeasurementConverterService::convertMeasurement($this->measurementUnit, $this->width),
+            'height' => MeasurementConverterService::convertMeasurement($this->measurementUnit, $this->height),
+            'depth' => MeasurementConverterService::convertMeasurement($this->measurementUnit, $this->depth),
```

Comment:
I think it would be neater to run the conversion inside getters and call them here, rather than only doing it when you encode the JSON